### PR TITLE
PR: Add themes to the terminal

### LINKED ISSUE #108: Add terminal color schemes
Terminal color schemes should match those of the editor and of the IPython console. I don't know if users should be able to change individually the Terminal color scheme, or if we should follow the behaviour of the IPython console. Any ideas @ccordoba12 @goanpeca ?

### PR DESCRIPTION
**Pull request by steff456**
The terminal uses the same color theme that the configured in spyder.

Fixes #108 

**Emacs**
image

**IDLE**
image

**Inkpot**
image

**Minimal**
image

**Monokai**
image

**Nightlion**
image

**Notepad++**
image

**Oblivion**
image

**Obsidian**
image

**Pastel**
image

**Pydev**
image

**Retta**
image

**Roboticket**
image

**Scintilla**
image

**Solarized Dark**
image

**Solarized Light**
image

**Spyder**
image

**Spyder dark**
image

**Sublime**
image

**Vibrant Ink**
image

**Zenburn**
image

Note: Right now the terminal is changing the theme when spyder is restarted. (issue https://github.com/spyder-ide/spyder/issues/10911)

### CODE REVIEW

**Comment by goanpeca** [comment]
Please add a couple of screenshots @steff456 :-)

**Comment by ccordoba12** [comment]
@steff456, please rebase this PR so we can review it.

File: spyder_terminal/widgets/style/themes.py [hunk 1]
```diff
@@ -0,0 +1,387 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Copyright (c) Spyder Project Contributors
+#
+# Licensed under the terms of the MIT License
+# (see LICENSE.txt for details)
+# -----------------------------------------------------------------------------
+ANSI_COLORS = {
```
Comment (goanpeca) [inline spyder_terminal/widgets/style/themes.py]: Question. The values for these colors come from the theme?

Some of them? all of them?

Comment (steff456) [reply spyder_terminal/widgets/style/themes.py]: None of them, the colors that I'm using from the configuration are in the `set_theme` method. This colors were created using a generator for terminal themes depending the ones that are in spyder.

Comment (ccordoba12) [reply spyder_terminal/widgets/style/themes.py]: > This colors were created using a generator for terminal themes depending the ones that are in spyder.

Is that generator a Python library? Or how could we use it in the future?

Comment (steff456) [reply spyder_terminal/widgets/style/themes.py]: Is an online generator, https://terminal.sexy/, which allows you to create a terminal color scheme that makes sense. But it is manual, so each time a new theme is added to spyder we need to create the ANSI colors for it manually.

**Comment by ccordoba12** [comment]
I'm going to manually review this as soon as I can to see how the themes look live.

File: spyder_terminal/confpage.py [hunk 2]
```diff
@@ -45,7 +45,7 @@ def setup_page(self):
         valid_shells = zip(valid_shells, valid_shells)
         default_option = 'cmd' if WINDOWS else 'bash'
         shell_combo = self.create_combobox(_("Select the shell interpreter:"),
-                                           valid_shells, 'shell', restart=True,
+                                           valid_shells, 'shell',
```
Comment (ccordoba12) [inline spyder_terminal/confpage.py]: Why did you remove `restart=True` from here? I mean, it seems this option requires a restart because I'm not getting a new shell after changing this option in our Preferences.

Comment (steff456) [reply spyder_terminal/confpage.py]: I thought this was another option, sorry hahaha
